### ISSUE TITLE
potential issues?

### ISSUE THREAD

**Issue by vanessadamario** [alphacsc/alphacsc#15]
First of all thank you a lot for sharing this code, it is extremely helpful. I found some issues after I cloned the repo from github and tried to run it.

- convolutional_dictionary_learning.py
line 336 - 362. Issue in the initialization of BatchCDL and GreedyCDL. By changing with super(BatchCDL, self) and super(GreedyCDL, self) respectively, I have solved the problem.
line 91. The documentation should be changed. There a conflict between the option 'chunks' for D_init and the one in init_dict.py, where at line 77 you make use of 'chunk' without s. 

Then I realized that for some runs the code, the fit method was not working and it raised this error
line 232, in learn_d_z_multi
    pobj, times, D_hat, z_hat = _batch_learn(greedy=True, **kwargs)
  File "/home/vanessa/src/neuralanalysis/neuralanalysis/alphacsc/learn_d_z_multi.py", line 316, in _batch_learn
    D_hat = np.concatenate([D_hat, new_atom[None]])
ValueError: all the input array dimensions except for the concatenation axis must match exactly

After a while I came up with a plausible reason, which I think is related to the initialization when using the 'chunk' option. The error pops up when the atom's dimension is smaller than the required one. 

- init_dict.py
in get_max_error_dict function, line 355. If I understood correctly, we take the patch based on the max reconstruction error. In the unravel_index of the vector, should the dimensions be 

n0, t0 = np.unravel_index(i0, (n_trials, n_times - n_times_atom)) instead of 
n0, t0 = np.unravel_index(i0, (n_trials, n_times)? 

This change fixed the bug. If I am not wrong you are selecting the best patch, but sometimes, with the previous reshape, t0 is higher than  (n_times - n_times_atom). So the following array 
   d0 = X[n0, :, t0:t0 + n_times_atom][None]
becomes smaller than the n_times_atom. Does it make sense?

Thank you a lot in advance!

**Comment by tomMoral** [comment]
@vanessadamario Thanks a lot for the issue!

It was indeed a bug in the unraveling of the indices in `init_dict.py`. I just pushed a fixed on master for this one. Sorry for the inconvenience.

I also fixed the documentation with the `chunk{s=>}` typo.

As for the `super` issue you report, I suppose you are using `python2.7`.
This version of python will not be maintained anymore starting January 2020 so we decided to only support `python3` for this repo.

Let us know if you are further difficulties!
